// oci-redis/lib/model/index.js
"use strict";
/**
 * Redis Service API
 * Use the Redis Service API to create and manage Redis clusters. A Redis cluster is a memory-based storage solution. For more information, see [OCI Caching Service with Redis](/iaas/Content/redis/home.htm).
 * OpenAPI spec version: 20220315
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateRedisClusterDetails = exports.SortOrder = exports.RedisClusterSummary = exports.RedisClusterCollection = exports.RedisCluster = exports.OperationType = exports.OperationStatus = exports.NodeCollection = exports.Node = exports.CreateRedisClusterDetails = exports.ChangeRedisClusterCompartmentDetails = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ChangeRedisClusterCompartmentDetails = __importStar(require("./change-redis-cluster-compartment-details"));
exports.ChangeRedisClusterCompartmentDetails = ChangeRedisClusterCompartmentDetails.ChangeRedisClusterCompartmentDetails;
const CreateRedisClusterDetails = __importStar(require("./create-redis-cluster-details"));
exports.CreateRedisClusterDetails = CreateRedisClusterDetails.CreateRedisClusterDetails;
const Node = __importStar(require("./node"));
exports.Node = Node.Node;
const NodeCollection = __importStar(require("./node-collection"));
exports.NodeCollection = NodeCollection.NodeCollection;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const RedisCluster = __importStar(require("./redis-cluster"));
exports.RedisCluster = RedisCluster.RedisCluster;
const RedisClusterCollection = __importStar(require("./redis-cluster-collection"));
exports.RedisClusterCollection = RedisClusterCollection.RedisClusterCollection;
const RedisClusterSummary = __importStar(require("./redis-cluster-summary"));
exports.RedisClusterSummary = RedisClusterSummary.RedisClusterSummary;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateRedisClusterDetails = __importStar(require("./update-redis-cluster-details"));
exports.UpdateRedisClusterDetails = UpdateRedisClusterDetails.UpdateRedisClusterDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
