// oci-capacitymanagement/lib/model/index.js
"use strict";
/**
 * OciControlCenterCp API
 * A description of the OciControlCenterCp API
 * OpenAPI spec version: 20231107
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchRequireInstruction = exports.PatchReplaceInstruction = exports.PatchRemoveInstruction = exports.PatchProhibitInstruction = exports.PatchMoveInstruction = exports.PatchMergeInstruction = exports.PatchInsertMultipleInstruction = exports.PatchInsertInstruction = exports.OccCapacityRequestComputeDetails = exports.UpdateOccCapacityRequestDetails = exports.UpdateOccAvailabilityCatalogDetails = exports.UpdateInternalOccCapacityRequestDetails = exports.SortOrder = exports.PatchOccCapacityRequestDetails = exports.PatchInstruction = exports.OccCustomerGroupSummary = exports.OccCustomerGroupCollection = exports.OccCustomerGroup = exports.OccCustomer = exports.OccCapacityRequestSummary = exports.OccCapacityRequestCollection = exports.OccCapacityRequestBaseDetails = exports.OccCapacityRequest = exports.OccAvailabilitySummary = exports.OccAvailabilityCollection = exports.OccAvailabilityCatalogSummary = exports.OccAvailabilityCatalogCollection = exports.OccAvailabilityCatalog = exports.Namespace = exports.MetadataDetails = exports.CreateOccCapacityRequestDetails = exports.CreateOccAvailabilityCatalogDetails = void 0;
const CreateOccAvailabilityCatalogDetails = __importStar(require("./create-occ-availability-catalog-details"));
exports.CreateOccAvailabilityCatalogDetails = CreateOccAvailabilityCatalogDetails.CreateOccAvailabilityCatalogDetails;
const CreateOccCapacityRequestDetails = __importStar(require("./create-occ-capacity-request-details"));
exports.CreateOccCapacityRequestDetails = CreateOccCapacityRequestDetails.CreateOccCapacityRequestDetails;
const MetadataDetails = __importStar(require("./metadata-details"));
exports.MetadataDetails = MetadataDetails.MetadataDetails;
const Namespace = __importStar(require("./namespace"));
exports.Namespace = Namespace.Namespace;
const OccAvailabilityCatalog = __importStar(require("./occ-availability-catalog"));
exports.OccAvailabilityCatalog = OccAvailabilityCatalog.OccAvailabilityCatalog;
const OccAvailabilityCatalogCollection = __importStar(require("./occ-availability-catalog-collection"));
exports.OccAvailabilityCatalogCollection = OccAvailabilityCatalogCollection.OccAvailabilityCatalogCollection;
const OccAvailabilityCatalogSummary = __importStar(require("./occ-availability-catalog-summary"));
exports.OccAvailabilityCatalogSummary = OccAvailabilityCatalogSummary.OccAvailabilityCatalogSummary;
const OccAvailabilityCollection = __importStar(require("./occ-availability-collection"));
exports.OccAvailabilityCollection = OccAvailabilityCollection.OccAvailabilityCollection;
const OccAvailabilitySummary = __importStar(require("./occ-availability-summary"));
exports.OccAvailabilitySummary = OccAvailabilitySummary.OccAvailabilitySummary;
const OccCapacityRequest = __importStar(require("./occ-capacity-request"));
exports.OccCapacityRequest = OccCapacityRequest.OccCapacityRequest;
const OccCapacityRequestBaseDetails = __importStar(require("./occ-capacity-request-base-details"));
exports.OccCapacityRequestBaseDetails = OccCapacityRequestBaseDetails.OccCapacityRequestBaseDetails;
const OccCapacityRequestCollection = __importStar(require("./occ-capacity-request-collection"));
exports.OccCapacityRequestCollection = OccCapacityRequestCollection.OccCapacityRequestCollection;
const OccCapacityRequestSummary = __importStar(require("./occ-capacity-request-summary"));
exports.OccCapacityRequestSummary = OccCapacityRequestSummary.OccCapacityRequestSummary;
const OccCustomer = __importStar(require("./occ-customer"));
exports.OccCustomer = OccCustomer.OccCustomer;
const OccCustomerGroup = __importStar(require("./occ-customer-group"));
exports.OccCustomerGroup = OccCustomerGroup.OccCustomerGroup;
const OccCustomerGroupCollection = __importStar(require("./occ-customer-group-collection"));
exports.OccCustomerGroupCollection = OccCustomerGroupCollection.OccCustomerGroupCollection;
const OccCustomerGroupSummary = __importStar(require("./occ-customer-group-summary"));
exports.OccCustomerGroupSummary = OccCustomerGroupSummary.OccCustomerGroupSummary;
const PatchInstruction = __importStar(require("./patch-instruction"));
exports.PatchInstruction = PatchInstruction.PatchInstruction;
const PatchOccCapacityRequestDetails = __importStar(require("./patch-occ-capacity-request-details"));
exports.PatchOccCapacityRequestDetails = PatchOccCapacityRequestDetails.PatchOccCapacityRequestDetails;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdateInternalOccCapacityRequestDetails = __importStar(require("./update-internal-occ-capacity-request-details"));
exports.UpdateInternalOccCapacityRequestDetails = UpdateInternalOccCapacityRequestDetails.UpdateInternalOccCapacityRequestDetails;
const UpdateOccAvailabilityCatalogDetails = __importStar(require("./update-occ-availability-catalog-details"));
exports.UpdateOccAvailabilityCatalogDetails = UpdateOccAvailabilityCatalogDetails.UpdateOccAvailabilityCatalogDetails;
const UpdateOccCapacityRequestDetails = __importStar(require("./update-occ-capacity-request-details"));
exports.UpdateOccCapacityRequestDetails = UpdateOccCapacityRequestDetails.UpdateOccCapacityRequestDetails;
const OccCapacityRequestComputeDetails = __importStar(require("./occ-capacity-request-compute-details"));
exports.OccCapacityRequestComputeDetails = OccCapacityRequestComputeDetails.OccCapacityRequestComputeDetails;
const PatchInsertInstruction = __importStar(require("./patch-insert-instruction"));
exports.PatchInsertInstruction = PatchInsertInstruction.PatchInsertInstruction;
const PatchInsertMultipleInstruction = __importStar(require("./patch-insert-multiple-instruction"));
exports.PatchInsertMultipleInstruction = PatchInsertMultipleInstruction.PatchInsertMultipleInstruction;
const PatchMergeInstruction = __importStar(require("./patch-merge-instruction"));
exports.PatchMergeInstruction = PatchMergeInstruction.PatchMergeInstruction;
const PatchMoveInstruction = __importStar(require("./patch-move-instruction"));
exports.PatchMoveInstruction = PatchMoveInstruction.PatchMoveInstruction;
const PatchProhibitInstruction = __importStar(require("./patch-prohibit-instruction"));
exports.PatchProhibitInstruction = PatchProhibitInstruction.PatchProhibitInstruction;
const PatchRemoveInstruction = __importStar(require("./patch-remove-instruction"));
exports.PatchRemoveInstruction = PatchRemoveInstruction.PatchRemoveInstruction;
const PatchReplaceInstruction = __importStar(require("./patch-replace-instruction"));
exports.PatchReplaceInstruction = PatchReplaceInstruction.PatchReplaceInstruction;
const PatchRequireInstruction = __importStar(require("./patch-require-instruction"));
exports.PatchRequireInstruction = PatchRequireInstruction.PatchRequireInstruction;
