// oci-capacitymanagement/lib/model/namespace.js
"use strict";
/**
 * OciControlCenterCp API
 * A description of the OciControlCenterCp API
 * OpenAPI spec version: 20231107
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Namespace = void 0;
/**
 * The name of the OCI service in consideration. For example, Compute, Exadata, and so on.
 *
 **/
var Namespace;
(function (Namespace) {
    Namespace["Compute"] = "COMPUTE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    Namespace["UnknownValue"] = "UNKNOWN_VALUE";
})(Namespace = exports.Namespace || (exports.Namespace = {}));
(function (Namespace) {
    function getJsonObj(obj) {
        return obj;
    }
    Namespace.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    Namespace.getDeserializedJsonObj = getDeserializedJsonObj;
})(Namespace = exports.Namespace || (exports.Namespace = {}));
