// oci-capacitymanagement/lib/model/occ-availability-summary.js
"use strict";
/**
 * OciControlCenterCp API
 * A description of the OciControlCenterCp API
 * OpenAPI spec version: 20231107
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OccAvailabilitySummary = void 0;
var OccAvailabilitySummary;
(function (OccAvailabilitySummary) {
    let ResourceType;
    (function (ResourceType) {
        ResourceType["ServerHw"] = "SERVER_HW";
        ResourceType["CapacityConstraint"] = "CAPACITY_CONSTRAINT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ResourceType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ResourceType = OccAvailabilitySummary.ResourceType || (OccAvailabilitySummary.ResourceType = {}));
    let WorkloadType;
    (function (WorkloadType) {
        WorkloadType["Generic"] = "GENERIC";
        WorkloadType["Row"] = "ROW";
        WorkloadType["UsProd"] = "US_PROD";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        WorkloadType["UnknownValue"] = "UNKNOWN_VALUE";
    })(WorkloadType = OccAvailabilitySummary.WorkloadType || (OccAvailabilitySummary.WorkloadType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    OccAvailabilitySummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    OccAvailabilitySummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(OccAvailabilitySummary = exports.OccAvailabilitySummary || (exports.OccAvailabilitySummary = {}));
