// oci-capacitymanagement/lib/model/update-occ-capacity-request-details.js
"use strict";
/**
 * OciControlCenterCp API
 * A description of the OciControlCenterCp API
 * OpenAPI spec version: 20231107
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateOccCapacityRequestDetails = void 0;
var UpdateOccCapacityRequestDetails;
(function (UpdateOccCapacityRequestDetails) {
    let RequestState;
    (function (RequestState) {
        RequestState["Submitted"] = "SUBMITTED";
        RequestState["Cancelled"] = "CANCELLED";
    })(RequestState = UpdateOccCapacityRequestDetails.RequestState || (UpdateOccCapacityRequestDetails.RequestState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateOccCapacityRequestDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UpdateOccCapacityRequestDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(UpdateOccCapacityRequestDetails = exports.UpdateOccCapacityRequestDetails || (exports.UpdateOccCapacityRequestDetails = {}));
