// oci-jms/lib/model/agent-type.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentType = void 0;
/**
 * The agent can be of two type, either Oracle Management Agent(OMA) or Oracle Cloud Agent(OCA).
 **/
var AgentType;
(function (AgentType) {
    AgentType["Oma"] = "OMA";
    AgentType["Oca"] = "OCA";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AgentType["UnknownValue"] = "UNKNOWN_VALUE";
})(AgentType = exports.AgentType || (exports.AgentType = {}));
(function (AgentType) {
    function getJsonObj(obj) {
        return obj;
    }
    AgentType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AgentType.getDeserializedJsonObj = getDeserializedJsonObj;
})(AgentType = exports.AgentType || (exports.AgentType = {}));
