// oci-jms/lib/model/application-installation-sort-by.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationInstallationSortBy = void 0;
/**
 * The field to sort application installations.
 **/
var ApplicationInstallationSortBy;
(function (ApplicationInstallationSortBy) {
    ApplicationInstallationSortBy["TimeFirstSeen"] = "timeFirstSeen";
    ApplicationInstallationSortBy["TimeLastSeen"] = "timeLastSeen";
    ApplicationInstallationSortBy["DisplayName"] = "displayName";
    ApplicationInstallationSortBy["InstallationPath"] = "installationPath";
    ApplicationInstallationSortBy["OsName"] = "osName";
    ApplicationInstallationSortBy["ApproximateJreCount"] = "approximateJreCount";
    ApplicationInstallationSortBy["ApproximateInstallationCount"] = "approximateInstallationCount";
    ApplicationInstallationSortBy["ApproximateManagedInstanceCount"] = "approximateManagedInstanceCount";
})(ApplicationInstallationSortBy = exports.ApplicationInstallationSortBy || (exports.ApplicationInstallationSortBy = {}));
(function (ApplicationInstallationSortBy) {
    function getJsonObj(obj) {
        return obj;
    }
    ApplicationInstallationSortBy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ApplicationInstallationSortBy.getDeserializedJsonObj = getDeserializedJsonObj;
})(ApplicationInstallationSortBy = exports.ApplicationInstallationSortBy || (exports.ApplicationInstallationSortBy = {}));
