// oci-jms/lib/model/drs-file.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DrsFile = void 0;
var DrsFile;
(function (DrsFile) {
    let ChecksumType;
    (function (ChecksumType) {
        ChecksumType["Sha256"] = "SHA256";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ChecksumType["UnknownValue"] = "UNKNOWN_VALUE";
    })(ChecksumType = DrsFile.ChecksumType || (DrsFile.ChecksumType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DrsFile.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DrsFile.getDeserializedJsonObj = getDeserializedJsonObj;
})(DrsFile = exports.DrsFile || (exports.DrsFile = {}));
