// oci-jms/lib/model/drs-file-sort-by.js
"use strict";
/**
 * Java Management Service API
 * API for the Java Management Service. Use this API to view, create, and manage Fleets.
 * OpenAPI spec version: 20210610
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DrsFileSortBy = void 0;
/**
 * The field used to sort the DRS details results.
 **/
var DrsFileSortBy;
(function (DrsFileSortBy) {
    DrsFileSortBy["BucketName"] = "bucketName";
    DrsFileSortBy["Namespace"] = "namespace";
    DrsFileSortBy["DrsFileKey"] = "drsFileKey";
    DrsFileSortBy["DrsFileName"] = "drsFileName";
    DrsFileSortBy["ChecksumType"] = "checksumType";
    DrsFileSortBy["IsDefault"] = "isDefault";
})(DrsFileSortBy = exports.DrsFileSortBy || (exports.DrsFileSortBy = {}));
(function (DrsFileSortBy) {
    function getJsonObj(obj) {
        return obj;
    }
    DrsFileSortBy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DrsFileSortBy.getDeserializedJsonObj = getDeserializedJsonObj;
})(DrsFileSortBy = exports.DrsFileSortBy || (exports.DrsFileSortBy = {}));
