// oci-osmanagementhub/lib/model/operation-types.js
"use strict";
/**
 * OS Management Hub API
 * Use the OS Management Hub API to manage and monitor updates and patches for the operating system environments in your private data centers through a single management console. For more information, see [Overview of OS Management Hub](https://docs.cloud.oracle.com/iaas/osmh/doc/overview.htm).

 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationTypes = void 0;
/**
 * Possible scheduled job operation types.
 **/
var OperationTypes;
(function (OperationTypes) {
    OperationTypes["InstallPackages"] = "INSTALL_PACKAGES";
    OperationTypes["UpdatePackages"] = "UPDATE_PACKAGES";
    OperationTypes["RemovePackages"] = "REMOVE_PACKAGES";
    OperationTypes["UpdateAll"] = "UPDATE_ALL";
    OperationTypes["UpdateSecurity"] = "UPDATE_SECURITY";
    OperationTypes["UpdateBugfix"] = "UPDATE_BUGFIX";
    OperationTypes["UpdateEnhancement"] = "UPDATE_ENHANCEMENT";
    OperationTypes["UpdateOther"] = "UPDATE_OTHER";
    OperationTypes["UpdateKspliceUserspace"] = "UPDATE_KSPLICE_USERSPACE";
    OperationTypes["UpdateKspliceKernel"] = "UPDATE_KSPLICE_KERNEL";
    OperationTypes["ManageModuleStreams"] = "MANAGE_MODULE_STREAMS";
    OperationTypes["SwitchModuleStream"] = "SWITCH_MODULE_STREAM";
    OperationTypes["AttachSoftwareSources"] = "ATTACH_SOFTWARE_SOURCES";
    OperationTypes["DetachSoftwareSources"] = "DETACH_SOFTWARE_SOURCES";
    OperationTypes["SyncManagementStationMirror"] = "SYNC_MANAGEMENT_STATION_MIRROR";
    OperationTypes["PromoteLifecycle"] = "PROMOTE_LIFECYCLE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationTypes["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationTypes = exports.OperationTypes || (exports.OperationTypes = {}));
(function (OperationTypes) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationTypes.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationTypes.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationTypes = exports.OperationTypes || (exports.OperationTypes = {}));
