// oci-osmanagementhub/lib/model/work-request-operation-type.js
"use strict";
/**
 * OS Management Hub API
 * Use the OS Management Hub API to manage and monitor updates and patches for the operating system environments in your private data centers through a single management console. For more information, see [Overview of OS Management Hub](https://docs.cloud.oracle.com/iaas/osmh/doc/overview.htm).

 * OpenAPI spec version: 20220901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestOperationType = void 0;
/**
 * Possible operation types.
 **/
var WorkRequestOperationType;
(function (WorkRequestOperationType) {
    WorkRequestOperationType["InstallPackages"] = "INSTALL_PACKAGES";
    WorkRequestOperationType["RemovePackages"] = "REMOVE_PACKAGES";
    WorkRequestOperationType["UpdatePackages"] = "UPDATE_PACKAGES";
    WorkRequestOperationType["UpdateAllPackages"] = "UPDATE_ALL_PACKAGES";
    WorkRequestOperationType["UpdateSecurity"] = "UPDATE_SECURITY";
    WorkRequestOperationType["UpdateBugfix"] = "UPDATE_BUGFIX";
    WorkRequestOperationType["UpdateEnhancement"] = "UPDATE_ENHANCEMENT";
    WorkRequestOperationType["UpdateOther"] = "UPDATE_OTHER";
    WorkRequestOperationType["UpdateKspliceKernel"] = "UPDATE_KSPLICE_KERNEL";
    WorkRequestOperationType["UpdateKspliceUserspace"] = "UPDATE_KSPLICE_USERSPACE";
    WorkRequestOperationType["EnableModuleStreams"] = "ENABLE_MODULE_STREAMS";
    WorkRequestOperationType["DisableModuleStreams"] = "DISABLE_MODULE_STREAMS";
    WorkRequestOperationType["SwitchModuleStream"] = "SWITCH_MODULE_STREAM";
    WorkRequestOperationType["InstallModuleProfiles"] = "INSTALL_MODULE_PROFILES";
    WorkRequestOperationType["RemoveModuleProfiles"] = "REMOVE_MODULE_PROFILES";
    WorkRequestOperationType["SetSoftwareSources"] = "SET_SOFTWARE_SOURCES";
    WorkRequestOperationType["ListPackages"] = "LIST_PACKAGES";
    WorkRequestOperationType["SetManagementStationConfig"] = "SET_MANAGEMENT_STATION_CONFIG";
    WorkRequestOperationType["SyncManagementStationMirror"] = "SYNC_MANAGEMENT_STATION_MIRROR";
    WorkRequestOperationType["UpdateManagementStationSoftware"] = "UPDATE_MANAGEMENT_STATION_SOFTWARE";
    WorkRequestOperationType["Update"] = "UPDATE";
    WorkRequestOperationType["ModuleActions"] = "MODULE_ACTIONS";
    WorkRequestOperationType["LifecyclePromotion"] = "LIFECYCLE_PROMOTION";
    WorkRequestOperationType["CreateSoftwareSource"] = "CREATE_SOFTWARE_SOURCE";
    WorkRequestOperationType["UpdateSoftwareSource"] = "UPDATE_SOFTWARE_SOURCE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    WorkRequestOperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(WorkRequestOperationType = exports.WorkRequestOperationType || (exports.WorkRequestOperationType = {}));
(function (WorkRequestOperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    WorkRequestOperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    WorkRequestOperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(WorkRequestOperationType = exports.WorkRequestOperationType || (exports.WorkRequestOperationType = {}));
