// oci-apmsynthetics/lib/model/index.js
"use strict";
/**
 * Application Performance Monitoring Synthetic Monitoring API
 * Use the Application Performance Monitoring Synthetic Monitoring API to query synthetic scripts and monitors. For more information, see [Application Performance Monitoring](https://docs.oracle.com/iaas/application-performance-monitoring/index.html).
 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicVantagePointCollection = exports.Protocol = exports.ProbeMode = exports.PrivateKey = exports.OnPremiseVpWorkerVersionDetails = exports.OnPremiseVantagePointWorkerType = exports.OnPremiseVantagePointWorkerStatus = exports.OnPremiseVantagePointSummary = exports.OnPremiseVantagePointCollection = exports.OnPremiseVantagePoint = exports.OAuthSchemes = exports.Node = exports.NetworkConfiguration = exports.MonitorTypes = exports.MonitorSummary = exports.MonitorStatusCountMap = exports.MonitorStatus = exports.MonitorScriptParameterInfo = exports.MonitorScriptParameter = exports.MonitorResultData = exports.MonitorResult = exports.MonitorConfiguration = exports.MonitorCollection = exports.Monitor = exports.MaintenanceWindowSchedule = exports.Link = exports.IdentityInfoDetails = exports.Header = exports.GeoSummary = exports.DvpStackDetails = exports.DnsTransportProtocol = exports.DnsRecordType = exports.DnsConfiguration = exports.DedicatedVantagePointSummary = exports.DedicatedVantagePointStatus = exports.DedicatedVantagePointCollection = exports.DedicatedVantagePoint = exports.CreateWorkerDetails = exports.CreateScriptDetails = exports.CreateOnPremiseVantagePointDetails = exports.CreateMonitorDetails = exports.CreateDedicatedVantagePointDetails = exports.ContentTypes = exports.ClientCertificateDetails = exports.ClientCertificate = exports.AvailableCapability = exports.AvailabilityConfiguration = exports.AggregatedNetworkDataResult = exports.AggregatedNetworkData = exports.AggregateNetworkDataDetails = void 0;
exports.ScriptedRestMonitorConfiguration = exports.ScriptedBrowserMonitorConfiguration = exports.RestMonitorConfiguration = exports.OracleRMStack = exports.NetworkMonitorConfiguration = exports.DnsTraceMonitorConfiguration = exports.DnsServerMonitorConfiguration = exports.DnsSecMonitorConfiguration = exports.BrowserMonitorConfiguration = exports.WorkersSummary = exports.WorkerSummary = exports.WorkerMonitorList = exports.WorkerCollection = exports.Worker = exports.VerifyText = exports.VantagePointNode = exports.VantagePointInfo = exports.VantagePointExecution = exports.UpdateWorkerDetails = exports.UpdateScriptDetails = exports.UpdateOnPremiseVantagePointDetails = exports.UpdateMonitorDetails = exports.UpdateDedicatedVantagePointDetails = exports.SortOrders = exports.ScriptSummary = exports.ScriptParameterInfo = exports.ScriptParameter = exports.ScriptCollection = exports.Script = exports.SchedulingPolicy = exports.RequestQueryParam = exports.RequestMethods = exports.RequestAuthenticationSchemesForScriptedRest = exports.RequestAuthenticationSchemes = exports.RequestAuthenticationDetails = exports.PublicVantagePointSummary = void 0;
const AggregateNetworkDataDetails = __importStar(require("./aggregate-network-data-details"));
exports.AggregateNetworkDataDetails = AggregateNetworkDataDetails.AggregateNetworkDataDetails;
const AggregatedNetworkData = __importStar(require("./aggregated-network-data"));
exports.AggregatedNetworkData = AggregatedNetworkData.AggregatedNetworkData;
const AggregatedNetworkDataResult = __importStar(require("./aggregated-network-data-result"));
exports.AggregatedNetworkDataResult = AggregatedNetworkDataResult.AggregatedNetworkDataResult;
const AvailabilityConfiguration = __importStar(require("./availability-configuration"));
exports.AvailabilityConfiguration = AvailabilityConfiguration.AvailabilityConfiguration;
const AvailableCapability = __importStar(require("./available-capability"));
exports.AvailableCapability = AvailableCapability.AvailableCapability;
const ClientCertificate = __importStar(require("./client-certificate"));
exports.ClientCertificate = ClientCertificate.ClientCertificate;
const ClientCertificateDetails = __importStar(require("./client-certificate-details"));
exports.ClientCertificateDetails = ClientCertificateDetails.ClientCertificateDetails;
const ContentTypes = __importStar(require("./content-types"));
exports.ContentTypes = ContentTypes.ContentTypes;
const CreateDedicatedVantagePointDetails = __importStar(require("./create-dedicated-vantage-point-details"));
exports.CreateDedicatedVantagePointDetails = CreateDedicatedVantagePointDetails.CreateDedicatedVantagePointDetails;
const CreateMonitorDetails = __importStar(require("./create-monitor-details"));
exports.CreateMonitorDetails = CreateMonitorDetails.CreateMonitorDetails;
const CreateOnPremiseVantagePointDetails = __importStar(require("./create-on-premise-vantage-point-details"));
exports.CreateOnPremiseVantagePointDetails = CreateOnPremiseVantagePointDetails.CreateOnPremiseVantagePointDetails;
const CreateScriptDetails = __importStar(require("./create-script-details"));
exports.CreateScriptDetails = CreateScriptDetails.CreateScriptDetails;
const CreateWorkerDetails = __importStar(require("./create-worker-details"));
exports.CreateWorkerDetails = CreateWorkerDetails.CreateWorkerDetails;
const DedicatedVantagePoint = __importStar(require("./dedicated-vantage-point"));
exports.DedicatedVantagePoint = DedicatedVantagePoint.DedicatedVantagePoint;
const DedicatedVantagePointCollection = __importStar(require("./dedicated-vantage-point-collection"));
exports.DedicatedVantagePointCollection = DedicatedVantagePointCollection.DedicatedVantagePointCollection;
const DedicatedVantagePointStatus = __importStar(require("./dedicated-vantage-point-status"));
exports.DedicatedVantagePointStatus = DedicatedVantagePointStatus.DedicatedVantagePointStatus;
const DedicatedVantagePointSummary = __importStar(require("./dedicated-vantage-point-summary"));
exports.DedicatedVantagePointSummary = DedicatedVantagePointSummary.DedicatedVantagePointSummary;
const DnsConfiguration = __importStar(require("./dns-configuration"));
exports.DnsConfiguration = DnsConfiguration.DnsConfiguration;
const DnsRecordType = __importStar(require("./dns-record-type"));
exports.DnsRecordType = DnsRecordType.DnsRecordType;
const DnsTransportProtocol = __importStar(require("./dns-transport-protocol"));
exports.DnsTransportProtocol = DnsTransportProtocol.DnsTransportProtocol;
const DvpStackDetails = __importStar(require("./dvp-stack-details"));
exports.DvpStackDetails = DvpStackDetails.DvpStackDetails;
const GeoSummary = __importStar(require("./geo-summary"));
exports.GeoSummary = GeoSummary.GeoSummary;
const Header = __importStar(require("./header"));
exports.Header = Header.Header;
const IdentityInfoDetails = __importStar(require("./identity-info-details"));
exports.IdentityInfoDetails = IdentityInfoDetails.IdentityInfoDetails;
const Link = __importStar(require("./link"));
exports.Link = Link.Link;
const MaintenanceWindowSchedule = __importStar(require("./maintenance-window-schedule"));
exports.MaintenanceWindowSchedule = MaintenanceWindowSchedule.MaintenanceWindowSchedule;
const Monitor = __importStar(require("./monitor"));
exports.Monitor = Monitor.Monitor;
const MonitorCollection = __importStar(require("./monitor-collection"));
exports.MonitorCollection = MonitorCollection.MonitorCollection;
const MonitorConfiguration = __importStar(require("./monitor-configuration"));
exports.MonitorConfiguration = MonitorConfiguration.MonitorConfiguration;
const MonitorResult = __importStar(require("./monitor-result"));
exports.MonitorResult = MonitorResult.MonitorResult;
const MonitorResultData = __importStar(require("./monitor-result-data"));
exports.MonitorResultData = MonitorResultData.MonitorResultData;
const MonitorScriptParameter = __importStar(require("./monitor-script-parameter"));
exports.MonitorScriptParameter = MonitorScriptParameter.MonitorScriptParameter;
const MonitorScriptParameterInfo = __importStar(require("./monitor-script-parameter-info"));
exports.MonitorScriptParameterInfo = MonitorScriptParameterInfo.MonitorScriptParameterInfo;
const MonitorStatus = __importStar(require("./monitor-status"));
exports.MonitorStatus = MonitorStatus.MonitorStatus;
const MonitorStatusCountMap = __importStar(require("./monitor-status-count-map"));
exports.MonitorStatusCountMap = MonitorStatusCountMap.MonitorStatusCountMap;
const MonitorSummary = __importStar(require("./monitor-summary"));
exports.MonitorSummary = MonitorSummary.MonitorSummary;
const MonitorTypes = __importStar(require("./monitor-types"));
exports.MonitorTypes = MonitorTypes.MonitorTypes;
const NetworkConfiguration = __importStar(require("./network-configuration"));
exports.NetworkConfiguration = NetworkConfiguration.NetworkConfiguration;
const Node = __importStar(require("./node"));
exports.Node = Node.Node;
const OAuthSchemes = __importStar(require("./o-auth-schemes"));
exports.OAuthSchemes = OAuthSchemes.OAuthSchemes;
const OnPremiseVantagePoint = __importStar(require("./on-premise-vantage-point"));
exports.OnPremiseVantagePoint = OnPremiseVantagePoint.OnPremiseVantagePoint;
const OnPremiseVantagePointCollection = __importStar(require("./on-premise-vantage-point-collection"));
exports.OnPremiseVantagePointCollection = OnPremiseVantagePointCollection.OnPremiseVantagePointCollection;
const OnPremiseVantagePointSummary = __importStar(require("./on-premise-vantage-point-summary"));
exports.OnPremiseVantagePointSummary = OnPremiseVantagePointSummary.OnPremiseVantagePointSummary;
const OnPremiseVantagePointWorkerStatus = __importStar(require("./on-premise-vantage-point-worker-status"));
exports.OnPremiseVantagePointWorkerStatus = OnPremiseVantagePointWorkerStatus.OnPremiseVantagePointWorkerStatus;
const OnPremiseVantagePointWorkerType = __importStar(require("./on-premise-vantage-point-worker-type"));
exports.OnPremiseVantagePointWorkerType = OnPremiseVantagePointWorkerType.OnPremiseVantagePointWorkerType;
const OnPremiseVpWorkerVersionDetails = __importStar(require("./on-premise-vp-worker-version-details"));
exports.OnPremiseVpWorkerVersionDetails = OnPremiseVpWorkerVersionDetails.OnPremiseVpWorkerVersionDetails;
const PrivateKey = __importStar(require("./private-key"));
exports.PrivateKey = PrivateKey.PrivateKey;
const ProbeMode = __importStar(require("./probe-mode"));
exports.ProbeMode = ProbeMode.ProbeMode;
const Protocol = __importStar(require("./protocol"));
exports.Protocol = Protocol.Protocol;
const PublicVantagePointCollection = __importStar(require("./public-vantage-point-collection"));
exports.PublicVantagePointCollection = PublicVantagePointCollection.PublicVantagePointCollection;
const PublicVantagePointSummary = __importStar(require("./public-vantage-point-summary"));
exports.PublicVantagePointSummary = PublicVantagePointSummary.PublicVantagePointSummary;
const RequestAuthenticationDetails = __importStar(require("./request-authentication-details"));
exports.RequestAuthenticationDetails = RequestAuthenticationDetails.RequestAuthenticationDetails;
const RequestAuthenticationSchemes = __importStar(require("./request-authentication-schemes"));
exports.RequestAuthenticationSchemes = RequestAuthenticationSchemes.RequestAuthenticationSchemes;
const RequestAuthenticationSchemesForScriptedRest = __importStar(require("./request-authentication-schemes-for-scripted-rest"));
exports.RequestAuthenticationSchemesForScriptedRest = RequestAuthenticationSchemesForScriptedRest.RequestAuthenticationSchemesForScriptedRest;
const RequestMethods = __importStar(require("./request-methods"));
exports.RequestMethods = RequestMethods.RequestMethods;
const RequestQueryParam = __importStar(require("./request-query-param"));
exports.RequestQueryParam = RequestQueryParam.RequestQueryParam;
const SchedulingPolicy = __importStar(require("./scheduling-policy"));
exports.SchedulingPolicy = SchedulingPolicy.SchedulingPolicy;
const Script = __importStar(require("./script"));
exports.Script = Script.Script;
const ScriptCollection = __importStar(require("./script-collection"));
exports.ScriptCollection = ScriptCollection.ScriptCollection;
const ScriptParameter = __importStar(require("./script-parameter"));
exports.ScriptParameter = ScriptParameter.ScriptParameter;
const ScriptParameterInfo = __importStar(require("./script-parameter-info"));
exports.ScriptParameterInfo = ScriptParameterInfo.ScriptParameterInfo;
const ScriptSummary = __importStar(require("./script-summary"));
exports.ScriptSummary = ScriptSummary.ScriptSummary;
const SortOrders = __importStar(require("./sort-orders"));
exports.SortOrders = SortOrders.SortOrders;
const UpdateDedicatedVantagePointDetails = __importStar(require("./update-dedicated-vantage-point-details"));
exports.UpdateDedicatedVantagePointDetails = UpdateDedicatedVantagePointDetails.UpdateDedicatedVantagePointDetails;
const UpdateMonitorDetails = __importStar(require("./update-monitor-details"));
exports.UpdateMonitorDetails = UpdateMonitorDetails.UpdateMonitorDetails;
const UpdateOnPremiseVantagePointDetails = __importStar(require("./update-on-premise-vantage-point-details"));
exports.UpdateOnPremiseVantagePointDetails = UpdateOnPremiseVantagePointDetails.UpdateOnPremiseVantagePointDetails;
const UpdateScriptDetails = __importStar(require("./update-script-details"));
exports.UpdateScriptDetails = UpdateScriptDetails.UpdateScriptDetails;
const UpdateWorkerDetails = __importStar(require("./update-worker-details"));
exports.UpdateWorkerDetails = UpdateWorkerDetails.UpdateWorkerDetails;
const VantagePointExecution = __importStar(require("./vantage-point-execution"));
exports.VantagePointExecution = VantagePointExecution.VantagePointExecution;
const VantagePointInfo = __importStar(require("./vantage-point-info"));
exports.VantagePointInfo = VantagePointInfo.VantagePointInfo;
const VantagePointNode = __importStar(require("./vantage-point-node"));
exports.VantagePointNode = VantagePointNode.VantagePointNode;
const VerifyText = __importStar(require("./verify-text"));
exports.VerifyText = VerifyText.VerifyText;
const Worker = __importStar(require("./worker"));
exports.Worker = Worker.Worker;
const WorkerCollection = __importStar(require("./worker-collection"));
exports.WorkerCollection = WorkerCollection.WorkerCollection;
const WorkerMonitorList = __importStar(require("./worker-monitor-list"));
exports.WorkerMonitorList = WorkerMonitorList.WorkerMonitorList;
const WorkerSummary = __importStar(require("./worker-summary"));
exports.WorkerSummary = WorkerSummary.WorkerSummary;
const WorkersSummary = __importStar(require("./workers-summary"));
exports.WorkersSummary = WorkersSummary.WorkersSummary;
const BrowserMonitorConfiguration = __importStar(require("./browser-monitor-configuration"));
exports.BrowserMonitorConfiguration = BrowserMonitorConfiguration.BrowserMonitorConfiguration;
const DnsSecMonitorConfiguration = __importStar(require("./dns-sec-monitor-configuration"));
exports.DnsSecMonitorConfiguration = DnsSecMonitorConfiguration.DnsSecMonitorConfiguration;
const DnsServerMonitorConfiguration = __importStar(require("./dns-server-monitor-configuration"));
exports.DnsServerMonitorConfiguration = DnsServerMonitorConfiguration.DnsServerMonitorConfiguration;
const DnsTraceMonitorConfiguration = __importStar(require("./dns-trace-monitor-configuration"));
exports.DnsTraceMonitorConfiguration = DnsTraceMonitorConfiguration.DnsTraceMonitorConfiguration;
const NetworkMonitorConfiguration = __importStar(require("./network-monitor-configuration"));
exports.NetworkMonitorConfiguration = NetworkMonitorConfiguration.NetworkMonitorConfiguration;
const OracleRMStack = __importStar(require("./oracle-rm-stack"));
exports.OracleRMStack = OracleRMStack.OracleRMStack;
const RestMonitorConfiguration = __importStar(require("./rest-monitor-configuration"));
exports.RestMonitorConfiguration = RestMonitorConfiguration.RestMonitorConfiguration;
const ScriptedBrowserMonitorConfiguration = __importStar(require("./scripted-browser-monitor-configuration"));
exports.ScriptedBrowserMonitorConfiguration = ScriptedBrowserMonitorConfiguration.ScriptedBrowserMonitorConfiguration;
const ScriptedRestMonitorConfiguration = __importStar(require("./scripted-rest-monitor-configuration"));
exports.ScriptedRestMonitorConfiguration = ScriptedRestMonitorConfiguration.ScriptedRestMonitorConfiguration;
