// oci-tenantmanagercontrolplane/lib/model/classic-subscription-environment-name.js
"use strict";
/**
 * Organizations API
 * Use the Organizations API to consolidate multiple OCI tenancies into an organization, and centrally manage your tenancies and organization resources. For more information, see [Organization Management Overview](/iaas/Content/General/Concepts/organization_management_overview.htm).
 * OpenAPI spec version: 20230401
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClassicSubscriptionEnvironmentName = void 0;
/**
 * TAS (Tenant Automation System) legacy environment this order orginated from.
 **/
var ClassicSubscriptionEnvironmentName;
(function (ClassicSubscriptionEnvironmentName) {
    ClassicSubscriptionEnvironmentName["Prod"] = "PROD";
    ClassicSubscriptionEnvironmentName["Preprod"] = "PREPROD";
    ClassicSubscriptionEnvironmentName["Preprod1"] = "PREPROD1";
    ClassicSubscriptionEnvironmentName["Pintlab"] = "PINTLAB";
    ClassicSubscriptionEnvironmentName["Miglab"] = "MIGLAB";
    ClassicSubscriptionEnvironmentName["Pool2"] = "POOL2";
    ClassicSubscriptionEnvironmentName["Pintlab2"] = "PINTLAB2";
    ClassicSubscriptionEnvironmentName["Mylab0"] = "MYLAB0";
    ClassicSubscriptionEnvironmentName["Mylab1"] = "MYLAB1";
    ClassicSubscriptionEnvironmentName["Mylab2"] = "MYLAB2";
    ClassicSubscriptionEnvironmentName["Mylab3"] = "MYLAB3";
    ClassicSubscriptionEnvironmentName["Mylab4"] = "MYLAB4";
    ClassicSubscriptionEnvironmentName["Mylab5"] = "MYLAB5";
})(ClassicSubscriptionEnvironmentName = exports.ClassicSubscriptionEnvironmentName || (exports.ClassicSubscriptionEnvironmentName = {}));
(function (ClassicSubscriptionEnvironmentName) {
    function getJsonObj(obj) {
        return obj;
    }
    ClassicSubscriptionEnvironmentName.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ClassicSubscriptionEnvironmentName.getDeserializedJsonObj = getDeserializedJsonObj;
})(ClassicSubscriptionEnvironmentName = exports.ClassicSubscriptionEnvironmentName || (exports.ClassicSubscriptionEnvironmentName = {}));
